packages/core/src/render3/jit/module.ts

```diff
@@ -202,6 +202,15 @@ export function isStandalone<T>(type: Type<T>) {
   return def !== null ? def.standalone : false;
 }
 
+export function generateStandaloneInDeclarationsError(type: Type<any>, location: string) {
+  const errorPrefix = `Unexpected "${stringifyForError(type)}" found in the "declarations" array`;
+  const errorSuffix = `"${
+      stringifyForError(
+          type)}" is marked as standalone and can't be declared in any NgModule - did you intend to import it instead (by adding it to the "imports" array)?`;
+  const error = `${errorPrefix} ${location}, ${errorSuffix}`;
+  return error;
```

Comment:
nit: small refactoring to drop "error" from const names (it's clear from the context) and split the suffix string to use string concatenation (so that there is no weird auto-formatting):
```suggestion
  const prefix = `Unexpected "${stringifyForError(type)}" found in the "declarations" array`;
  const suffix =
      `"${stringifyForError(type)}" is marked as standalone and can't be declared ` +
      `in any NgModule - did you intend to import it instead (by adding it to the "imports" array)?`;
  return `${prefix} ${location}, ${suffix}`;
```